README.md

```diff
@@ -132,3 +132,97 @@ method: http
 http:
    port: 3903
 ```
+
+## TLS configuration
+
+You can use exporter_exporter with TLS to encrypt the traffic, and at the
+same time enforce strong mutual authentication between the nodes and the
+prometheus server.
+
+Note that `-web.tls.verify` will accept *any* certificate signed by the
+`-web.tls.ca`, so you need to create a separate CA for this purpose - or use
+a self-signed certificate, which acts as its own CA.
+
+Here is a simple configuration example, using one key/cert for the
+prometheus server and one key/cert shared between all the remote nodes.
+Firstly, create the keys and certs:
+
+```
+openssl req -x509 -newkey rsa:1024 -keyout prom_node_key.pem -out prom_node_cert.pem -days 29220 -nodes -subj /commonName=prom_node/
+openssl req -x509 -newkey rsa:1024 -keyout prometheus_key.pem -out prometheus_cert.pem -days 29220 -nodes -subj /commonName=prometheus/
+```
+
+Create an `/etc/prometheus/ssl/` directory on the prometheus server and all
+the remote nodes.  Install both cert.pem files everywhere.  It is safe for
+them to be world-readable.
+
+Install `prom_node_key.pem` only on the nodes, and set file permissions to
+protect it so that only exporter_exporter can read it.  Similarly, install
+`prometheus_key.pem` only on the prometheus server, and set permissions so
+that only the prometheus process can read it.
+
+Configuration for exporter_exporter on the nodes (here it also disables
+plain HTTP):
+
+```
+EXPEXP_FLAGS='-web.listen-address= -web.tls.listen-address=:9998
+ -web.tls.cert=/etc/prometheus/ssl/prom_node_cert.pem
+ -web.tls.key=/etc/prometheus/ssl/prom_node_key.pem
+ -web.tls.ca=/etc/prometheus/ssl/prometheus_cert.pem
+ -web.tls.verify'
+```
+
+To test it, create a temporary entry for `prom_node` in your `/etc/hosts`
+file on the prometheus server:
+
+```
+# the target you want to test
+192.0.2.1   prom_node
```

Comment:
Rather than suggest editing the hosts file, please remove this and update the curl example to use         `--resolve <host:port:address>`

Reply:
Nice find!